tools/whitespace-format.sh

```diff
@@ -0,0 +1,95 @@
+#!/usr/bin/env bash
+#
+#  Script to fix whitespace issues: trailing whitespace and DOS carriage returns.
+#
+#  Licensed to the Apache Software Foundation (ASF) under one
+#  or more contributor license agreements.  See the NOTICE file
+#  distributed with this work for additional information
+#  regarding copyright ownership.  The ASF licenses this file
+#  to you under the Apache License, Version 2.0 (the
+#  "License"); you may not use this file except in compliance
+#  with the License.  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+set -e
+
+# Get the repository root
+REPO_ROOT=$(cd $(dirname $0) && git rev-parse --show-toplevel)
+
+EXCLUDE_PATTERNS=(
+    "lib/yamlcpp"
+    "lib/Catch2"
+    "lib/systemtap"
+    ".gold:"
+    ".test_input"
+)
+
+GREP_EXCLUDE_ARGS=""
+for pattern in "${EXCLUDE_PATTERNS[@]}"; do
+    GREP_EXCLUDE_ARGS="${GREP_EXCLUDE_ARGS} | grep -F -v '${pattern}'"
```

Comment:
This variable seems unused.  I thought maybe grep will use it, but couldn't find info on that.  Also it's not exported.  Maybe just remove this?